dataprofiler/profilers/utils.py

```diff
@@ -455,26 +488,18 @@ def find_diff_of_dates(stat1, stat2):
     :type stat1: datetime.datetime object
     :param stat2: the second statistical input
     :type stat2: datetime.datetime object
-    :return: Difference in stats
-    :rtype: str
+    :return: difference in stats
+    :rtype: Union[List, str]
     """
-    # We can use find_diff_of_numbers since datetime objects
-    # can be compared and subtracted naturally
-    diff = find_diff_of_numbers(stat1, stat2)
-    if isinstance(diff, str):
-        return diff
-    if isinstance(diff, list):
-        # Deal with the case where one stat is None
-        return [None if i is None else i.strftime("%x %X") for i in diff]
-
-    # Must be timedelta object
-    if diff.days >= 0:
-        return "+" + str(diff)
-
-    return "-" + str(abs(diff))
+    if stat1 is None or stat2 is None:
+        return [i if i is None else i.strftime("%x %X") for i in [stat1, stat2]]
+    elif stat1 != stat2:
+        diff = stat1 - stat2
+        return ("+" if diff.days >= 0 else "-") + str(abs(diff))
+    return "unchanged"
```

Comment:
any reason for removing the 
`diff = find_diff_of_numbers(stat1, stat2)`
and handling this here?

Reply:
`find_diff_of_numbers()` only takes in ints and floats. If I make it also take in `datetime`, then mypy complains because ints/floats can't be subtracted with `datetime`.

Reply:
Fair, should we fix numbers to take in any type that can have a subtraction operation? Not sure if that's actually feasible. Otherwise, can do as is but fix the above.

Reply:
Found this:
https://stackoverflow.com/questions/37669222/how-can-i-hint-that-a-type-is-comparable-with-typing
```python
from __future__ import annotations

from abc import abstractmethod
from typing import MutableSequence, Protocol, TypeVar


class Comparable(Protocol):
    """Protocol for annotating comparable types."""

    @abstractmethod
    def __lt__(self: CT, other: CT) -> bool:
        pass


CT = TypeVar("CT", bound=Comparable)

def comparison_sort(s: MutableSequence[CT]) -> None:
    pass


comparison_sort([1, 2, 3])  # OK
comparison_sort([1.0, 2.0, 3.0])  # OK
comparison_sort(["42", "420", "2137"])  # OK
comparison_sort([1, 2, "3"])  # mypy error 
```

Could we alter `Comparable` name and abstractions to use `__eq__ `and `__sub__`. which would allow a datetime to go into diff_of_numbers?

Maybe that's not appropriate either bc it would take in anything with that.

Reply:
I added this and it seems to work fine. The only thing is that I need to make `__sub__` return `Any` because subtracting `datetime` results in a `timedelta`.